project-koku/koku
# COST-2715: Include report_period_id in tag based rates delete.

Fixes [COST-2715](https://issues.redhat.com/browse/COST-2715)


Changes proposed in this PR:
* add report period id to tag based rates sql delete


Testing Instructions:
I discovered this while working on currency support. 
1. You can create multiple different ocp sources with this command if you want:
```
curl -s -L https://raw.githubusercontent.com/myersCody/cody_notes/main/koku_features/currency_support/nise_yamls/OCP_on_yaml/ocp_commands.sh | bash
```
2. Add different Cost models. 
I have some fancy postman logic I built here:
- https://github.com/myersCody/cody_notes/tree/main/koku_features/currency_support/nise_yamls/OCP_on_yaml/cost_models

If you used the command in step one, all you need to do is update the currency var here in postman to create the 3 different cost models:
https://github.com/myersCody/cody_notes/blob/main/koku_features/currency_support/nise_yamls/OCP_on_yaml/cost_models/postman_prerequest.js#L3

The 3 currencies are `AUD`, `JPY` & `EUR`

3. After the cost model are added run this SQL:
```
postgres=# select
        distinct(usage_start),
        source_uuid,
        max(cluster_id),
        SUM(((infrastructure_usage_cost->>'cpu'::"text"))::numeric) as infra_usage,
        SUM(((infrastructure_usage_cost->>'memory'::"text"))::numeric) as infra_memory,
        SUM(((infrastructure_usage_cost->>'storage'::"text"))::numeric) as infra_storage,
        SUM(((infrastructure_usage_cost->>'cpu'::"text"))::numeric) as sup_usage,
        SUM(((infrastructure_usage_cost->>'memory'::"text"))::numeric) as sup_memory,
        SUM(((infrastructure_usage_cost->>'storage'::"text"))::numeric) as sup_storage
    from reporting_ocpusagelineitem_daily_summary WHERE
        usage_start >= '2022-06-1'
    group by usage_start, source_uuid order by usage_start
;
```
You will see cost for different sources now:
```
usage_start   | 2022-06-03
source_uuid   | 2d085606-e8fa-4a91-9ef8-755c05cb36e2
max           | ocp-cluster-JPY
infra_usage   | 60.00000000000000000
infra_memory  | 0.000000000000000
infra_storage | 0.000000000000000
sup_usage     | 60.00000000000000000
sup_memory    | 0.000000000000000
sup_storage   | 0.000000000000000
-[ RECORD 8 ]-+-------------------------------------
usage_start   | 2022-06-03
source_uuid   | 7188a517-5653-4482-bc57-c60b4a8f88a8
max           | ocp-cluster-EUR
infra_usage   | 60.00000000000000000
infra_memory  | 0.000000000000000
infra_storage | 0.000000000000000
sup_usage     | 60.00000000000000000
sup_memory    | 0.000000000000000
sup_storage   | 0.000000000000000
```
